File: solutions/clojure/anagram/1/src/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn -true-anagram [word candidate]
  (when-not (= word candidate)
    (and
     (or (= word (str/lower-case word))
         (= word (str/capitalize word)))
     (= (sort (str/lower-case word))
        (sort (str/lower-case candidate))))))

(defn anagrams-for [word prospect-list]
  (filter #(-true-anagram word %) prospect-list))

File: solutions/julia/anagram/1/anagram.jl
function sort_lowercase(word::AbstractString)
    sort(split(lowercase(word), ""))
end

function detect_anagrams(subject::AbstractString, candidates::AbstractArray)
    subjectsorted = sort_lowercase(subject)
    anagrams = [
        candidate for candidate in candidates
        if subjectsorted == sort_lowercase(candidate) &&
        lowercase(candidate) != lowercase(subject)]
    anagrams
end
